platforms/documentation/docs/src/samples/writing-tasks/custom-test-task/common/plugin/src/main/java/com/example/CustomTest.java

```diff
@@ -0,0 +1,104 @@
+package com.example;
+
+import org.gradle.api.DefaultTask;
+import org.gradle.api.provider.Property;
+import org.gradle.api.tasks.TaskAction;
+import org.gradle.api.tasks.Input;
+import org.gradle.api.tasks.options.Option;
+import org.gradle.api.tasks.testing.GroupTestEventReporter;
+import org.gradle.api.tasks.testing.TestEventReporter;
+import org.gradle.api.tasks.testing.TestEventReporterFactory;
+import org.gradle.api.tasks.testing.TestOutputEvent;
+
+import javax.inject.Inject;
+import java.time.Instant;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Random;
+
+abstract class CustomTest extends DefaultTask {
+    @Inject
+    public CustomTest() {
+    }
+
+    @Input
+    @Option(option="fail", description = "Tells the task to demonstrate failures.")
+    public abstract Property<Boolean> getFail();
```

Comment:
💭 Extra complexity in the custom task that newcomers will have to realize isn't necessary when defining their own.  A system prop would be better for the toggle.

Reply:
I think this looks more idiomatic and I don't have to deal with wiring system properties for CC compatibility.